### PR TITLE
pupa identifiers

### LINKED ISSUE #274: Distinct vote events, despite same motion, org, and date
From time to time it happens in the Illinois legislature that a motion will be brought forward for a vote, it will fail, and then it will be brought back again on the same day for a revote.

For example http://ilga.gov/legislation/votehistory.asp?DocNum=2963&DocTypeID=HB&LegID=104486&GAID=14&SessionID=91&GA=100&SpecSess=

The `pupa` vote importer complains that these votes are dupes. Thoughts on how to not have that happen.

### PR DESCRIPTION
**Pull request by fgregg**
This PR will add an ability to for the scraper to specify a pupa only identifier for vote events and events. Sometimes the scraper can know that two ocd objects are the same or different, even though the ocd attributes are not clear. 

This will close #274.

I decided to take a [GenericRelations](https://docs.djangoproject.com/en/1.8/ref/contrib/contenttypes/#generic-relations) approach. I think it's pretty reasonable.

@jamesturk @mileswwatkins , I'd like feedback on the general approach. 

I'd also like thoughts on the name. I don't really like `pupa_id` but couldn't find a better name.

To do

- [x] tests for events

### CODE REVIEW

**Comment by jamesturk** [comment]
OK, took some time to think this through.  I think it makes sense.

- after waffling on if it should be here or in py-ocd, keeping it here does seem right
- I'm good w/ the approach, the GenericForeignKey shouldn't incur much expense since this is only done at import time

I have a few implementation suggestions, I'll make those inline, but in general I think I'm a 👍

**Comment by jamesturk** [comment]
unrelated (except that we should ensure this patch matches) I noticed it had been a while since flake8 had been run on pupa, currently working on adding that to the travis job so we can stay consistent in the future & cleaning up the rough edges

**Comment by fgregg** [comment]
cool. should we do flake8 cleanup as a separately from this PR?

**Comment by fgregg** [comment]
okay, I rebased on your flake8 changes and brought my code into flake8 compliance.

**Comment by fgregg** [comment]
@jamesturk, bump

**Comment by jamesturk** [comment]
ah thanks- any thoughts on the code tweaks?  I'm glad to merge today then make them (or we can hold off on the https://github.com/opencivicdata/pupa/pull/280#discussion-diff-118170946L283 piece too if you think it doesn't make sense)

**Comment by fgregg** [comment]
sorry, I don't see the tweaks?

File: pupa/importers/events.py [hunk 1]
```diff
@@ -38,14 +39,21 @@ def __init__(self, jurisdiction_id, org_importer, person_importer, bill_importer
...
     def get_object(self, event):
-        spec = {
-            'name': event['name'],
-            'description': event['description'],
-            'start_time': event['start_time'],
-            'end_time': event['end_time'],
-            'timezone': event['timezone'],
-            'jurisdiction_id': self.jurisdiction_id
-        }
+        if event.get('pupa_id'):
```
Comment (fgregg) [inline pupa/importers/events.py]: Good idea? Where should things like that live?

**Comment by jamesturk** [comment]
hm, maybe just in the parent class?

**Comment by fgregg** [comment]
okay, I think I've addressed your feedback. good points all!

File: pupa/importers/base.py [hunk 2]
```diff
@@ -281,6 +284,14 @@ def import_item(self, data):
                                                                            self.model_class))
             self._create_related(obj, related, self.related_models)
 
+        if pupa_id:
```
Comment (fgregg) [inline pupa/importers/base.py]: cool, I didn't know about the defaults arg!

**Comment by jamesturk** [comment]
👍

**Pull request merged**